server/src/main/java/org/opensearch/indices/ShardLimitValidator.java

```diff
@@ -140,6 +171,25 @@ private static int getTotalShardCount(ClusterState state, Index index) {
         return indexMetadata.getNumberOfShards() * (1 + indexMetadata.getNumberOfReplicas());
     }
 
+    /**
+     * Returns true if the cluster.ignore_dot_indexes property is set to true and the index name
+     * starts with '.' else false.
+     *
+     * @param indexName  The index which needs to be validated.
+     */
+    private boolean validateDotIndex(String indexName) {
+        return this.ignoreDotIndexes && indexName.charAt(0) == '.';
+    }
```

Comment:
lets not club both the logic in this method looks confusing to read, just have `indexName.charAt(0) == '.'`